Refit/RequestBuilderImplementation.cs

```diff
@@ -16,6 +16,11 @@ namespace Refit
 {
     partial class RequestBuilderImplementation : IRequestBuilder
     {
+        static readonly ISet<HttpMethod> bodylessMethods = new HashSet<HttpMethod>
+        {
+            HttpMethod.Get,
+            HttpMethod.Head
```

Comment:
Are there other body-less methods? How about OPTIONS?

It may be better to white-list the ones with a body? PUT/PATCH/POST? Not sure about DELETE.

@darylmiller?

Reply:
I dug into the source of `HttpWebRequest`:

```csharp
    internal static KnownHttpVerb Get = new KnownHttpVerb("GET", false, true, false, false);
    internal static KnownHttpVerb Connect = new KnownHttpVerb("CONNECT", false, true, true, false);
    internal static KnownHttpVerb Head = new KnownHttpVerb("HEAD", false, true, false, true);
    internal static KnownHttpVerb Put = new KnownHttpVerb("PUT", true, false, false, false);
    internal static KnownHttpVerb Post = new KnownHttpVerb("POST", true, false, false, false);
    internal static KnownHttpVerb MkCol = new KnownHttpVerb("MKCOL", false, false, false, false);
```
The second boolean parameter is `contentBodyNotAllowed`

Reply:
Good enough for me.

Reply:
I also ran the following:

```csharp
await Task.WhenAll(
	typeof(HttpMethod).GetProperties(BindingFlags.Public | BindingFlags.Static)
		.Select(p => (HttpMethod)p.GetValue(null))
		.Select(async method =>
		{
			var request = new HttpRequestMessage
			{
				RequestUri = new Uri("http://httpbin.org/anything"),
				Method = method,
				Content = new StringContent("blah")
			};
			request.Content.Headers.TryAddWithoutValidation("Content-Type", "DERP");
			bool error = false;
			try
			{
				await new HttpClient().SendAsync(request);
			}
			catch (ProtocolViolationException)
			{
				error = true;		
			}
			return new { method.Method, error };
		})
	)
```

Output:

Method | error
-- | --
GET | True
PUT | False
POST | False
DELETE | False
HEAD | True
OPTIONS | False
TRACE | False

Reply:
So yeah, I agree that it's weird that those two are the only ones it doesn't like, but that seems to be the way it is.

Reply:
We don't really have OPTIONS either but someone can easily add it. FWIW, may be worthwhile adding the rest of those methods to the Attributes.cs?

Reply:
OPTIONS may make sense, but CONNECT is used to set up two way communication (usually when using SSL). It's not something that would typically be used directly for an API call, so I don't think it makes sense to add it.

I could add support for` [Options]`, but is this the right PR for it and are we sure we have a use case?

Reply:
If you want the spec's opinion, you can send a body with any request.  However, some methods are allowed to use the body, some are not. HEAD, GET and DELETE being the ones where the body is not allowed to mean anything.